src/ConstructionBase.jl

```diff
@@ -43,13 +43,9 @@ end
 
 getproperties(o::NamedTuple) = o
 getproperties(o::Tuple) = o
-@generated function getproperties(obj)
-    fnames = fieldnames(obj)
-    fvals = map(fnames) do fname
-        Expr(:call, :getproperty, :obj, QuoteNode(fname))
-    end
-    fvals = Expr(:tuple, fvals...)
-    :(NamedTuple{$fnames}($fvals))
+@inline function getproperties(obj::T) where {T}
```

Comment:
Did you have to do a bit of tinkering for this to infer or did it work right away?

Reply:
Do you know if the `@inline` is important?